### PR TITLE
Add ping plugin, cleanup and make README more verbose

### PR DESCRIPTION
**Pull request by jackharrhy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
 
 ## Running locally
 
+1. Clone the project by running `git@github.com:MUNComputerScienceSociety/Automata.git`, and change into the directory by running `cd ./Automata`
+1. Create the directory `mounted_plugins` within the project by running `mkdir ./mounted_plugins`
```
Comment (nint8835) [inline README.md]:
```suggestion
2. Create the directory `mounted_plugins` within the project by running `mkdir ./mounted_plugins`
```

File: README.md [hunk 2]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
 1. Copy `.env.dist` to `.env`
```
Comment (nint8835) [inline README.md]:
```suggestion
3. Copy `.env.dist` to `.env`
```

File: README.md [hunk 3]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
-2. Fill out the required information in the `.env`
+1. Fill out the required information in the `.env`
```
Comment (nint8835) [inline README.md]:
```suggestion
4. Fill out the required information in the `.env`
```

File: README.md [hunk 4]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
    - At the moment, the only required environment variable required is `AUTOMATA_TOKEN`, which is a Discord token, which you can see how to get [here](https://discordpy.readthedocs.io/en/latest/discord.html)
-3. Start the containers by running `docker-compose up -d`
+1. Create the network `web` by running `docker network create web`
```
Comment (nint8835) [inline README.md]:
```suggestion
5. Create the network `web` by running `docker network create web`
```

File: README.md [hunk 5]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
+1. Start the containers by running `docker-compose up -d`
```
Comment (nint8835) [inline README.md]:
```suggestion
6. Start the containers by running `docker-compose up -d`
```

File: README.md [hunk 6]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
 
 ## Developing your own plugins
 
+Automata is built around the [discord.py](https://discordpy.readthedocs.io/en/latest/) framework, therefore the plugins make heavy use of its decorators to abstract most of the complexity behind the scenes.
+
 1. Create the folder `mounted_plugins` if it doesn't already exist.
    - `plugins` is baked into the image when it is built, so editing files there won't have an effect
-2. Create a new plugin within the `mounted_plugins` folder
+1. Create a new plugin within the `mounted_plugins` folder
```
Comment (nint8835) [inline README.md]:
```suggestion
2. Create a new plugin within the `mounted_plugins` folder
```

File: README.md [hunk 7]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
    1. Create a new [Jigsaw plugin manifest](https://jigsaw.readthedocs.io/en/latest/plugin.json.html)
-      - You can use `plugins/PrivateChannels/plugin.json` as an example
-   2. Create a new plugin
-      - You can use `plugins/PrivateChannels/__init__.py` as an example
-3. Start the bot using the instructions from [Running locally](#running-locally)
+      - You can use `plugins/Ping/plugin.json` as an example
+   1. Create a new plugin
```
Comment (nint8835) [inline README.md]:
```suggestion
   2. Create a new plugin
```

File: README.md [hunk 8]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
+      - You can use `plugins/Ping/__init__.py` as an example
+1. Start the bot using the instructions from [Running locally](#running-locally)
```
Comment (nint8835) [inline README.md]:
```suggestion
2. Start the bot using the instructions from [Running locally](#running-locally)
```

File: README.md [hunk 9]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
 
 When you make changes to your plugins, restart the Automata container using `docker-compose restart automata`
 
 ## Developing the bot core and built-in plugins
 
 1. Edit the `docker-compose.yml`, replacing the `image: muncs/automata` line for the automata container with `build: .`
-2. Edit the bot core or the plugins as you wish
-3. Start the container, forcing a rebuild of the image using `docker-compose up -d --build`
+1. Edit the bot core or the plugins as you wish
```
Comment (nint8835) [inline README.md]:
```suggestion
2. Edit the bot core or the plugins as you wish
```

File: README.md [hunk 10]
```diff
@@ -7,29 +7,34 @@ Discord bot handling the management of the MUN Computer Science Society Discord
...
+1. Start the container, forcing a rebuild of the image using `docker-compose up -d --build`
```
Comment (nint8835) [inline README.md]:
```suggestion
3. Start the container, forcing a rebuild of the image using `docker-compose up -d --build`
```

File: plugins/Ping/__init__.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+from discord.ext import commands
+
+from Plugin import AutomataPlugin
+from Globals import mongo_client
+
+
+class Ping(AutomataPlugin):
+    """Pong"""
+
+    def __init__(self, manifest, bot: commands.Bot):
```
Comment (nint8835) [inline plugins/Ping/__init__.py]: Unnecessary constructor override

File: plugins/Ping/__init__.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
+import discord
+from discord.ext import commands
+
+from Plugin import AutomataPlugin
+from Globals import mongo_client
```
Comment (nint8835) [inline plugins/Ping/__init__.py]: Unnecessary import

File: README.md (hunk 3 above)
Comment (jackharrhy) [reply README.md]: 81148aac6e3d58a2fe90a03887e4494a2e120b49

File: README.md (hunk 9 above)
Comment (jackharrhy) [reply README.md]: 81148aac6e3d58a2fe90a03887e4494a2e120b49

File: README.md (hunk 10 above)
Comment (jackharrhy) [reply README.md]: 81148aac6e3d58a2fe90a03887e4494a2e120b49

File: README.md (hunk 8 above)
Comment (jackharrhy) [reply README.md]: 81148aac6e3d58a2fe90a03887e4494a2e120b49

File: plugins/Ping/__init__.py (hunk 11 above)
Comment (jackharrhy) [reply plugins/Ping/__init__.py]: 7782f11242edc556cb36e6249427d8d1d9518800

File: plugins/Ping/__init__.py (hunk 12 above)
Comment (jackharrhy) [reply plugins/Ping/__init__.py]: 7782f11242edc556cb36e6249427d8d1d9518800

**Pull request merged**